drivers/dma/dma_stm32_bdma.c

```diff
@@ -0,0 +1,927 @@
+	.reload		 = bdma_stm32_reload,
+	.config		 = bdma_stm32_configure,
+	.start		 = bdma_stm32_start,
+	.stop		 = bdma_stm32_stop,
+	.get_status	 = bdma_stm32_get_status,
+};
+
+#ifdef CONFIG_DMAMUX_STM32
+#define BDMA_STM32_OFFSET_INIT(index)			\
+	.offset = DT_INST_PROP(index, dma_offset),
+#else
+#define BDMA_STM32_OFFSET_INIT(index)
+#endif /* CONFIG_DMAMUX_STM32 */
+
+#define BDMA_STM32_INIT_DEV(index)					\
+static struct bdma_stm32_channel					\
+	bdma_stm32_channels_##index[BDMA_STM32_##index##_CHANNEL_COUNT];\
+									\
+const struct bdma_stm32_config bdma_stm32_config_##index = {		\
+	.pclken = { .bus = DT_INST_CLOCKS_CELL(index, bus),		\
+		    .enr = DT_INST_CLOCKS_CELL(index, bits) },		\
+	.config_irq = bdma_stm32_config_irq_##index,			\
+	.base = DT_INST_REG_ADDR(index),				\
+	.support_m2m = DT_INST_PROP(index, st_mem2mem),			\
+	.max_channels = BDMA_STM32_##index##_CHANNEL_COUNT,		\
+	.channels = bdma_stm32_channels_##index,			\
+	BDMA_STM32_OFFSET_INIT(index)					\
+};									\
+									\
+static struct bdma_stm32_data bdma_stm32_data_##index = {		\
+};									\
+									\
+DEVICE_DT_INST_DEFINE(index,							\
+		    &bdma_stm32_init,						\
+		    NULL,							\
+		    &bdma_stm32_data_##index, &bdma_stm32_config_##index,	\
+		    PRE_KERNEL_1, CONFIG_KERNEL_INIT_PRIORITY_DEFAULT,		\
+		    &dma_funcs)
+
+#define BDMA_STM32_DEFINE_IRQ_HANDLER(bdma, chan)			\
+static void bdma_stm32_irq_##bdma##_##chan(const struct device *dev)	\
+{									\
+	bdma_stm32_irq_handler(dev, chan);				\
+}
+
+
+#define BDMA_STM32_IRQ_CONNECT(bdma, chan)				\
+	do {								\
+		IRQ_CONNECT(DT_INST_IRQ_BY_IDX(bdma, chan, irq),	\
+			    DT_INST_IRQ_BY_IDX(bdma, chan, priority),	\
+			    bdma_stm32_irq_##bdma##_##chan,		\
+			    DEVICE_DT_INST_GET(bdma), 0);		\
+		irq_enable(DT_INST_IRQ_BY_IDX(bdma, chan, irq));	\
+	} while (false)
+
+
+#if DT_NODE_HAS_STATUS(DT_DRV_INST(0), okay)
+
+BDMA_STM32_DEFINE_IRQ_HANDLER(0, 0);
+BDMA_STM32_DEFINE_IRQ_HANDLER(0, 1);
+BDMA_STM32_DEFINE_IRQ_HANDLER(0, 2);
+BDMA_STM32_DEFINE_IRQ_HANDLER(0, 3);
+BDMA_STM32_DEFINE_IRQ_HANDLER(0, 4);
+#if DT_INST_IRQ_HAS_IDX(0, 5)
+BDMA_STM32_DEFINE_IRQ_HANDLER(0, 5);
+#if DT_INST_IRQ_HAS_IDX(0, 6)
+BDMA_STM32_DEFINE_IRQ_HANDLER(0, 6);
+#if DT_INST_IRQ_HAS_IDX(0, 7)
+BDMA_STM32_DEFINE_IRQ_HANDLER(0, 7);
+#endif /* DT_INST_IRQ_HAS_IDX(0, 5) */
+#endif /* DT_INST_IRQ_HAS_IDX(0, 6) */
+#endif /* DT_INST_IRQ_HAS_IDX(0, 7) */
+
+static void bdma_stm32_config_irq_0(const struct device *dev)
+{
+	ARG_UNUSED(dev);
+
+	BDMA_STM32_IRQ_CONNECT(0, 0);
+	BDMA_STM32_IRQ_CONNECT(0, 1);
+	BDMA_STM32_IRQ_CONNECT(0, 2);
+	BDMA_STM32_IRQ_CONNECT(0, 3);
+	BDMA_STM32_IRQ_CONNECT(0, 4);
+#if DT_INST_IRQ_HAS_IDX(0, 5)
+	BDMA_STM32_IRQ_CONNECT(0, 5);
+#if DT_INST_IRQ_HAS_IDX(0, 6)
+	BDMA_STM32_IRQ_CONNECT(0, 6);
+#if DT_INST_IRQ_HAS_IDX(0, 7)
+	BDMA_STM32_IRQ_CONNECT(0, 7);
+#endif /* DT_INST_IRQ_HAS_IDX(0, 5) */
+#endif /* DT_INST_IRQ_HAS_IDX(0, 6) */
+#endif /* DT_INST_IRQ_HAS_IDX(0, 7) */
```

Comment:
Ditto

Reply:
Implemented.

Reply:
Also implemented for `dma_stm32.c` in seperate commit.